bundler/lib/dependabot/bundler/file_parser.rb

```diff
@@ -32,9 +32,28 @@ def parse
         dependency_set.dependencies
       end
 
-      sig { returns(PackageManagerBase) }
+      sig { returns(Ecosystem) }
+      def ecosystem
+        @ecosystem ||= T.let(
+          Ecosystem.new(
+            name: ECOSYSTEM,
+            package_manager: package_manager,
+            language: language
+          ),
+          T.nilable(Ecosystem)
+        )
+      end
+
       def package_manager
-        PackageManager.new(bundler_version)
+        return @package_manager if defined?(@package_manager)
+
+        @package_manager = PackageManager.new(bundler_version)
+      end
```

Comment:
We are returning existings package_manager solutions without changing the functionality. Just we changed structure for upcoming implementation of metric collections. 

Details: https://github.com/github/dependabot-updates/pull/7207